### PR TITLE
Add useResponsiveVariant as hydration-safe way to select desktop or mobile

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Replace most usages of `useBreakpoint` with `useResponsiveVariant`

Main use case if hydration-compatible selection of mobile vs desktop variant

## Justify why they are needed

Main driving goal was to make page header smaller - we don't want to keep desktop menu in DOM on mobile and we don't want to render it either

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4021?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4021** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4021?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4019** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4019?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4018** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4011** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4011?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/utils/useResponsiveVariant.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { useEffect, useState } from 'react'
+import { type Level } from 'ui'
+import { useBreakpoint } from './useBreakpoint/useBreakpoint'
+
+// Lazy responsive rendering, we don't want hydration errors
+// and we don't want user-agent dependency for server side (mainly because it's not compatible with SSG)
+export const useResponsiveVariant = (breakpoint: Level) => {
+  const [variant, setVariant] = useState<'desktop' | 'mobile' | 'initial'>('initial')
```
Comment (guilhermespopolin) [inline apps/store/src/utils/useResponsiveVariant.ts:8]: nit:
```suggestion
  const [variant, setVariant] = useState<'desktop' | 'mobile' | 'unkown'>('unkown')
```
What do you think about this change?

**Review by guilhermespopolin — APPROVED** [review]
Nice refactoring 🚀

File: apps/store/src/utils/useResponsiveVariant.ts (hunk 1 above)
Comment (alebedev) [reply apps/store/src/utils/useResponsiveVariant.ts:8]: I think 'unknown' is a bit misleading, we always know variant after initial client rendering.  Not a fan of 'initial' either, would be happy to improve it

Comment (guilhermespopolin) [reply apps/store/src/utils/useResponsiveVariant.ts:8]: I see. I was thinking on that 'initial' value as something we only use on the server as the placeholder for the period we can't know which variant we're dealing with. That's why 'unknow'.

**Comment by alebedev** [comment]
### Merge activity

* **Mar 6, 9:45 AM EST**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4021).

**Pull request merged**